.buildkite/pipeline.yml

```diff
@@ -25,15 +25,16 @@ steps:
      - ./rpm-packages/*
    branches: "!master"
  - commands:
-   - nix-build ./nix -A binaries -o binaries
-   - ./scripts/test-binaries.sh binaries/bin
+   - buildkite-agent artifact download "binaries/bin/*" . --step "build and package"
+   - chmod +x ./binaries/bin/*
+   - nix-build tests/tezos-binaries.nix --no-out-link --arg path-to-binaries ./binaries/bin
    label: test binaries
```

Comment:
Can we change it to something like "test nix-built binaries"? And I suppose "build and package" should changed to something like "build with nix and package". And why do we package before building with docker? Or do we build with docker here? Below I see "build using docker".

I think this file requires some refactoring:
1. Labels are a bit outdated.
2. It'd be nice to start each step with `label` as we do in most repos.
3. I'd also add some comments probably, but that depends on how you will update the labels.
4. Time to update the copyright year :wink: 

I don't know if it's a good idea to put all this refactoring into this PR. Feel free to do it fully here or only partially here and then making another PR.

Reply:
I did refactoring in this PR in a separate commit

Reply:
While we are at it, is `NIX_PATH` used?

Reply:
yes, there is plenty of `nix run nixpkgs.*`

Reply:
But we also have `sources.json` that specifies version of `nixpkgs`. Why are we using `NIX_PATH` in addition to that?

Reply:
I'd say that it's historically happened